### ISSUE TITLE
The green add button just dissapeared.

### ISSUE THREAD

**Issue by todorpashkulov** [themoeway/yomitan#818]
**Description**
The extension was working fine and all of a sudden the green add to anki button disappeared. No idea what broke it. Anki connect is working. 

**Browser version**
Firefox 124.0.2

**Yomitan version**
Yomitan 24.4.7.0
Anki Version ⁨2.1.48

**Exported settings file**
[yomitan-settings-2024-04-14-23-42-13.json](https://github.com/themoeway/yomitan/files/14972555/yomitan-settings-2024-04-14-23-42-13.json)
![image](https://github.com/themoeway/yomitan/assets/44975797/8c802b06-4ca2-4167-942a-74659eedca3d)

**Comment by todorpashkulov** [comment]
I found the problem when the setting.
```
Check for card duplicates
When a card is detected as a duplicate, the add buttons will be disabled.
```
Is enabled it detects all cards as duplicates and always hides the add buttons no idea if the problem is because of my anki version.

**Comment by Kuuuube** [comment]
It isn't supposed to hide the add buttons ever (disable != hide). But it's a bit misleading now. In previous versions it would gray them out but now they show as a red color when theres a duplicate and still let you add a dupe if you want.

**Comment by Kuuuube** [comment]
I cannot replicate this issue using the latest anki version. If you have a way to test on latest or can update that would be helpful.

**Comment by kanjieater** [comment]
I'm on a recent version (not yet latest) and am running into the same thing. If I disable the duplicate check at a Deck level it reenables the add button.
バージョン ⁨23.12.1 (1a1d4d54)⁩
Python 3.9.15 Qt 5.15.2 PyQt 5.15.5

**Comment by Kuuuube** [comment]
Are there any yomitan errors in console (F12 -> Console) when this happens?

**Comment by kanjieater** [comment]
> Are there any yomitan errors in console (F12 -> Console) when this happens?

![image](https://github.com/themoeway/yomitan/assets/32607317/66a0931d-a4e4-4256-9e68-50be38833142)
None relating to this functionality - these exist whether it's on or off
![image](https://github.com/themoeway/yomitan/assets/32607317/489bffd7-cc37-4168-9d32-89d14b06e84a)

**Comment by joseph-allen** [comment]
Not sure if intended, but I can get into the HTML and see the actions are there, and set to hidden and disabled. If I remove these qualities manually adding works, so the bug may be in the logic setting hidden and disabled.

**Comment by pavliuk-ii** [comment]
Same issue on my android phone (firefox 124.2.0, yomitan 24.4.7.0, ankidroid 2.17.6, ankiconnect mobile 1.12)
Disabling "Check for card duplicates" makes the buttons display, but then I have no idea if I already had the word added before which defeats the purpose of the function at all.

**Comment by joseph-allen** [comment]
I can't find the option for "Check for card duplicates" @pavliuk-ii, is that a Anki, Anki connect or Yomitan?

**Comment by pavliuk-ii** [comment]
@joseph-allen It's in the anki section of yomitan settings.
![image](https://github.com/themoeway/yomitan/assets/71011278/da2600fc-8148-4e32-ac63-3075ce42c988)

**Comment by joseph-allen** [comment]
Ah fantastic, it's an advanced setting in my settings but that's all I needed. And toggling that did make the add button return for me too

**Comment by pavliuk-ii** [comment]
Yeah, the biggest issue with disabling it is that when it's off you can't see which words you have mined before and which not and you end up with new duplicates every time

**Comment by Kuuuube** [comment]
Got the entire anki collection, yomitan settings, and yomitan dicts of someone with this issue. And I still can't replicate the issue.

I can try to look into fixing it but no guarantees since I currently have no way to test.

If anyone can figure out more specifically what is causing this that would be helpful. Or if anyone with this issue knows how to use git and can bisect and find the bad commit.

**Comment by Kuuuube** [comment]
Also @eloyrobillard since it's most likely #693 that is causing this if you have any insight here.

**Comment by genednzl** [comment]
Having the same problem, reset all settings on Yomitan but the green button still not showing up.
Even the scanning stopped working so really hoping someone can fix this.
Gonna try to disable the duplicate card option when im home see if that works like for everyone else

**Comment by eloyrobillard** [comment]
@Kuuuube I'll look into this asap

> Not sure if intended, but I can get into the HTML and see the actions are there, and set to hidden and disabled. If I remove these qualities manually adding works, so the bug may be in the logic setting hidden and disabled.

that sounds plausible indeed

**Comment by Kuuuube** [comment]
Please try the latest pre release https://github.com/themoeway/yomitan/releases/tag/24.4.15.0 and see if you can replicate the bug there.

If it is present, check the console and it should now display some log about what when wrong.

**Comment by YashaNoos** [comment]
I have the same problem as op. Just tested 24.4.15.0 and sadly the green add button didn't came back.

Kiwi Browser 124.0.6327.2
Android 14
Anki connect android 1.12
Anki 23.12.1

**Comment by Mansive** [comment]
Encountered this issue on Kiwi Browser 8265119645, Yomitan 24.4.15.0. I've pasted the console logs below:
```
log.js:138 Yomitan (development build) 24.4.15.0 has encountered a problem.
Originating URL: chrome-extension://ifgadlclliipkebdbniaddamapenidgc/popup.html?type=terms&query=%E7%9C%81%E7%95%A5&wildcards=off
ExtensionError: Anki error: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $
	at com.google.gson.JsonParser.parseReader(JsonParser.java:66)
	at com.google.gson.JsonParser.parseString(JsonParser.java:47)
	at com.kamwithk.ankiconnectandroid.routing.AnkiAPIRouting.findRouteHandleError(AnkiAPIRouting.java:101)
	at com.kamwithk.ankiconnectandroid.routing.APIHandler.chooseAPI(APIHandler.java:33)
	at com.kamwithk.ankiconnectandroid.routing.RouteHandler.get(RouteHandler.java:70)
	at fi.iki.elonen.router.RouterNanoHTTPD$DefaultStreamHandler.post(RouterNanoHTTPD.java:100)
	at fi.iki.elonen.router.RouterNanoHTTPD$UriResource.process(RouterNanoHTTPD.java:386)
	at fi.iki.elonen.router.RouterNanoHTTPD$UriRouter.process(RouterNanoHTTPD.java:596)
	at fi.iki.elonen.router.RouterNanoHTTPD.serve(RouterNanoHTTPD.java:671)
	at fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:945)
	at fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:192)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $
	at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1597)
	at com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1404)
	at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:542)
	at com.google.gson.stream.JsonReader.peek(JsonReader.java:425)
	at com.google.gson.JsonParser.parseReader(JsonParser.java:61)
	... 11 more

    at AnkiConnect._invoke (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/comm/anki-connect.js:441:31)
    at async AnkiConnect.canAddNotesWithErrorDetail (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/comm/anki-connect.js:138:24)
    at async Backend.detectDuplicateNotes (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/background/backend.js:547:16)
    at async Backend.partitionAddibleNotes (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/background/backend.js:557:31)
    at async Backend._onApiGetAnkiNoteInfo (chrome-extension://ifgadlclliipkebdbniaddamapenidgc/js/background/backend.js:583:47)
Data: {
    "action": "canAddNotesWithErrorDetail",
    "params": {
        "notes": [
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            },
            {
                "fields": {},
                "tags": [
                    "yomitan"
                ],
                "deckName": "",
                "modelName": "",
                "options": {
                    "allowDuplicate": false,
                    "duplicateScope": "collection",
                    "duplicateScopeOptions": {
                        "deckName": null,
                        "checkChildren": false,
                        "checkAllModels": false
                    }
                }
            }
        ]
    },
    "status": 200,
    "apiError": "com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $\n\tat com.google.gson.JsonParser.parseReader(JsonParser.java:66)\n\tat com.google.gson.JsonParser.parseString(JsonParser.java:47)\n\tat com.kamwithk.ankiconnectandroid.routing.AnkiAPIRouting.findRouteHandleError(AnkiAPIRouting.java:101)\n\tat com.kamwithk.ankiconnectandroid.routing.APIHandler.chooseAPI(APIHandler.java:33)\n\tat com.kamwithk.ankiconnectandroid.routing.RouteHandler.get(RouteHandler.java:70)\n\tat fi.iki.elonen.router.RouterNanoHTTPD$DefaultStreamHandler.post(RouterNanoHTTPD.java:100)\n\tat fi.iki.elonen.router.RouterNanoHTTPD$UriResource.process(RouterNanoHTTPD.java:386)\n\tat fi.iki.elonen.router.RouterNanoHTTPD$UriRouter.process(RouterNanoHTTPD.java:596)\n\tat fi.iki.elonen.router.RouterNanoHTTPD.serve(RouterNanoHTTPD.java:671)\n\tat fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:945)\n\tat fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:192)\n\tat java.lang.Thread.run(Thread.java:1012)\nCaused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 14 path $\n\tat com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1597)\n\tat com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1404)\n\tat com.google.gson.stream.JsonReader.doPeek(JsonReader.java:542)\n\tat com.google.gson.stream.JsonReader.peek(JsonReader.java:425)\n\tat com.google.gson.JsonParser.parseReader(JsonParser.java:61)\n\t... 11 more\n"
}

Issues can be reported at https://github.com/themoeway/yomitan/issues
```

**Comment by Kuuuube** [comment]
This may be related to the android ankiconnect + kiwi browser issues https://github.com/KamWithK/AnkiconnectAndroid/issues/50

If anyone who is experiencing this issue on pc can get a log that would be helpful.

**Comment by Kuuuube** [comment]
@eloyrobillard would it be possible to add a fallback using the previous "check for duplicates" ankiconnect api call if the current one fails? canAddNotesWithErrorDetail seems to be too new in some cases

**Comment by Kuuuube** [comment]
Error log from a user on PC, sent on tmw discord. [errorlog.txt](https://github.com/themoeway/yomitan/files/14985431/errorlog.txt)

Appears to be an ankiconnect error caused by `canAddNotesWithErrorDetail` and an outdated version of ankiconnect.

**Comment by Kuuuube** [comment]
For any users experiencing this on pc make sure you have ankiconnect updated to the latest version. It appears that some older anki versions may not be receiving new ankiconnect updates so you may have to update anki as well.

**Comment by kanjieater** [comment]
> For any users experiencing this on pc make sure you have ankiconnect updated to the latest version. It appears that some older anki versions may not be receiving new ankiconnect updates so you may have to update anki as well.

I can confirm that updating anki connect allows me to reenable the duplicate setting, same as before this release.

However, it seems the new code does disable adding duplicates at a deck level like it did previously. 
If you accidentally double click the "add card button that then becomes the add duplicate button", it adds a duplicate to your deck, which isn't how it used to work. It used to just disable adding a duplicate card once it was in the deck.

**Comment by eloyrobillard** [comment]
> @eloyrobillard would it be possible to add a fallback using the previous "check for duplicates" ankiconnect api call if the current one fails? canAddNotesWithErrorDetail seems to be too new in some cases

yes, I'll do that

**Comment by eloyrobillard** [comment]
I've prepared a fix in #827
